packages/wdio-browserstack-service/src/accessibility-handler.ts

```diff
@@ -216,8 +243,26 @@ class _AccessibilityHandler {
 
         try {
             const shouldScanScenario = shouldScanTestForAccessibility(featureData?.name, pickleData.name, this._accessibilityOptions, world, true)
+            if (isAppAccessibilityAutomationSession(this._accessibility, this.isAppAutomate)) {
+                this._testMetadata[uniqueId] = {
+                    scanTestForAccessibility : shouldScanScenario,
+                    accessibilityScanStarted : true
+                }
+                this._testMetadata[uniqueId].accessibilityScanStarted = shouldScanScenario
+                if (this._sessionId) {
+                    /* For case with multiple tests under one browser, before hook of 2nd test should change this map value */
+                    AccessibilityHandler._a11yScanSessionMap[this._sessionId] = shouldScanScenario
+                }
+                Listener.setTestRunAccessibilityVar(this._accessibility && shouldScanScenario)
+                if (shouldScanScenario) {
+                    BStackLogger.info('AppAutomate test case execution has started.')
+                }
+                return
+            }
             const isPageOpened = await this.checkIfPageOpened(this._browser, uniqueId, shouldScanScenario)
```

Comment:
Can we add !isApp condition on isPageOpened only rather than duplicating the code.

More better, we can removed for web A11y as well not needed now `isPageOpened`?

+ @amaanbs